pkg/operator/controller/externaldns/pod.go

```diff
@@ -194,6 +194,10 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 		args = append(args, "--ignore-hostname-annotation")
 	}
 
+	if b.externalDNS.Spec.AWSRoleARN != nil {
+		args = append(args, fmt.Sprintf("--aws-assume-role=%s", *b.externalDNS.Spec.AWSRoleARN))
```

Comment:
I'm wondering whether we should expose [the external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html) as the API knob too. ExternalDNS has [a dedicated flag](https://github.com/PG2000/external-dns/blob/91aaf6803df58463c0380bedbf647e230062b510/pkg/apis/externaldns/types.go#L428) for it. I see that we didn't bother with in [C-I-O](https://github.com/openshift/cluster-ingress-operator/pull/928).

Reply:
That's a good point, I hadn't heard of the ExternalID until now. But, if we didn't bother with it in CIO, I wonder if we should bother with it here. I can point it out to the Shared VPC folks so they are aware something like this exists.